src/main/java/io/cdap/plugin/gcp/gcs/sink/GCSBatchSink.java

```diff
@@ -386,6 +398,50 @@ public void validate(FailureCollector collector) {
       }
     }
 
+    //This method validated the pattern of CMEK Key resource ID.
+    private void validateCmekKey(FailureCollector failureCollector) {
+      try {
+        CryptoKeyName cmekKeyName = CryptoKeyName.parse(cmekKey);
+        Boolean isServiceAccountFilePath = isServiceAccountFilePath();
+        Credentials credentials = null;
+        try {
+          credentials = getServiceAccount() == null ?
+            null : GCPUtils.loadServiceAccountCredentials(getServiceAccount(), isServiceAccountFilePath);
+        } catch (Exception e) {
+          /*Ignoring the exception because we don't want to highlight cmek key if an exception occurs while
+          loading credentials*/
+          return;
+        }
+        Storage storage = GCPUtils.getStorage(getProject(), credentials);
+        Bucket bucket = null;
+        try {
+          bucket = storage.get(getBucket());
+        } catch (StorageException e) {
+          /* Ignoring the exception because we don't want the validation to fail if there is an exception getting
+            the bucket information either because the service account used during validation is the CDF service account
+            which is different than the service account that will be used at runtime (the dataproc service account)
+            (assuming the user has auto-detect for the service account) */
+          return;
+        }
+        if (bucket == null) {
+          if (Strings.isNullOrEmpty(location)) {
+            location = "US";
+          }
+          String cmekKeyLocation = cmekKeyName.getLocation();
+          if (!cmekKeyLocation.equals(location)) {
+            failureCollector.addFailure(String.format("CMEK key '%s' is in location '%s' while the GCS bucket will " +
+                                                        "be created in location '%s'.", cmekKey,
+                                                      cmekKeyLocation, location)
+              , "Modify the CMEK key or bucket location to be the same")
+              .withConfigProperty(NAME_CMEK_KEY);
+          }
+        }
+      } catch (ValidationException e) {
```

Comment:
Nothing in the try block throws a ValidationException. You should not have a try-catch for this entire method. As mentioned before, this should only be around the part that parses the key.